In [84]:
import pandas as pd
import matplotlib.pyplot as plt

In [103]:
full_df = pd.read_csv("Advanced.csv", sep=",")
college_df = pd.read_csv("nba_college.csv", sep=",")
player_df = pd.read_csv("all_seasons.csv", sep=",")

In [104]:
# filter so that only specific columns are used
player_df = player_df[["player_name", "team_abbreviation", "age", "player_height", "player_weight", "college", "country", "draft_year", "draft_round", "gp", "pts", "reb", "ast"]]
display(player_df)

player_name team_abbreviation   age  player_height  player_weight  \
0      Randy Livingston               HOU  22.0         193.04      94.800728   
1      Gaylon Nickerson               WAS  28.0         190.50      86.182480   
2          George Lynch               VAN  26.0         203.20     103.418976   
3        George McCloud               LAL  30.0         203.20     102.058200   
4          George Zidek               DEN  23.0         213.36     119.748288   
...                 ...               ...   ...            ...            ...   
12839       Joel Embiid               PHI  29.0         213.36     127.005760   
12840   John Butler Jr.               POR  20.0         213.36      86.182480   
12841      John Collins               ATL  25.0         205.74     102.511792   
12842      Jericho Sims               NYK  24.0         208.28     113.398000   
12843    JaMychal Green               GSW  33.0         205.74     102.965384   

                     college   country draft_year draft_round  gp   pts   reb  \
0            Louisiana State       USA       1996           2  64   3.9   1.5   
1      Northwestern Oklahoma       USA       1994           2   4   3.8   1.3   
2             North Carolina       USA       1993           1  41   8.3   6.4   
3              Florida State       USA       1989           1  64  10.2   2.8   
4                       UCLA       USA       1995           1  52   2.8   1.7   
...                      ...       ...        ...         ...  ..   ...   ...   
12839                 Kansas  Cameroon       2014           1  66  33.1  10.2   
12840          Florida State       USA  Undrafted   Undrafted  19   2.4   0.9   
12841            Wake Forest       USA       2017           1  71  13.1   6.5   
12842                  Texas       USA       2021           2  52   3.4   4.7   
12843                Alabama       USA  Undrafted   Undrafted  57   6.4   3.6   

       ast  
0      2.4  
1      0.3  
2      1.9  
3      1.7  
4      0.3  
...    ...  
12839  4.2  
12840  0.6  
12841  1.2  
12842  0.5  
12843  0.9  

[12844 rows x 13 columns]

In [105]:
# drop missing values by deleting the column because it exists in another dataset
player_df = player_df.drop("college", axis=1)

In [106]:
# remove duplicates
duplicated = player_df.drop_duplicates()

In [108]:
# subset the dataset into various dataframes

# filter to get only USA players
usa_players = player_df[(player_df["country"] == "USA")]
display(usa_players)

# filter to get players that have more than 20 points per game
high_scorers = player_df[(player_df["pts"] > 20)]
display(high_scorers)

player_name team_abbreviation   age  player_height  player_weight  \
0      Randy Livingston               HOU  22.0         193.04      94.800728   
1      Gaylon Nickerson               WAS  28.0         190.50      86.182480   
2          George Lynch               VAN  26.0         203.20     103.418976   
3        George McCloud               LAL  30.0         203.20     102.058200   
4          George Zidek               DEN  23.0         213.36     119.748288   
...                 ...               ...   ...            ...            ...   
12838      Joe Wieskamp               TOR  23.0         198.12      92.986360   
12840   John Butler Jr.               POR  20.0         213.36      86.182480   
12841      John Collins               ATL  25.0         205.74     102.511792   
12842      Jericho Sims               NYK  24.0         208.28     113.398000   
12843    JaMychal Green               GSW  33.0         205.74     102.965384   

      country draft_year draft_round  gp   pts  reb  ast  
0         USA       1996           2  64   3.9  1.5  2.4  
1         USA       1994           2   4   3.8  1.3  0.3  
2         USA       1993           1  41   8.3  6.4  1.9  
3         USA       1989           1  64  10.2  2.8  1.7  
4         USA       1995           1  52   2.8  1.7  0.3  
...       ...        ...         ...  ..   ...  ...  ...  
12838     USA       2021           2   9   1.0  0.4  0.3  
12840     USA  Undrafted   Undrafted  19   2.4  0.9  0.6  
12841     USA       2017           1  71  13.1  6.5  1.2  
12842     USA       2021           2  52   3.4  4.7  0.5  
12843     USA  Undrafted   Undrafted  57   6.4  3.6  0.9  

[10721 rows x 12 columns]

player_name team_abbreviation   age  player_height  player_weight  \
7            Glen Rice               CHH  30.0         203.20      99.790240   
8       Glenn Robinson               MIL  24.0         200.66     106.594120   
9           Grant Hill               DET  24.0         203.20     102.058200   
18     Hakeem Olajuwon               HOU  34.0         213.36     115.665960   
22         Gary Payton               SEA  28.0         193.04      86.182480   
...                ...               ...   ...            ...            ...   
12822     Jaylen Brown               BOS  26.0         198.12     101.151016   
12825     Jayson Tatum               BOS  25.0         203.20      95.254320   
12829     Jerami Grant               POR  29.0         203.20      95.254320   
12834     Jimmy Butler               MIA  33.0         200.66     104.326160   
12839      Joel Embiid               PHI  29.0         213.36     127.005760   

        country draft_year draft_round  gp   pts   reb  ast  
7           USA       1989           1  79  26.8   4.0  2.0  
8           USA       1994           1  80  21.1   6.3  3.1  
9           USA       1994           1  80  21.4   9.0  7.3  
18      Nigeria       1984           1  78  23.2   9.2  3.0  
22          USA       1990           1  82  21.8   4.6  7.1  
...         ...        ...         ...  ..   ...   ...  ...  
12822       USA       2016           1  67  26.6   6.9  3.5  
12825       USA       2017           1  74  30.1   8.8  4.6  
12829       USA       2014           2  63  20.5   4.5  2.4  
12834       USA       2011           1  64  22.9   5.9  5.3  
12839  Cameroon       2014           1  66  33.1  10.2  4.2  

[701 rows x 12 columns]

Indicates that 10,721 rows out of 12,844 rows are USA athletes.

Also shows that out of 12,844 rows, only 701 are top scorers with over 20 points per game.

In [112]:
# sort data in various ways

# sort so that it decreases by age first and then increases by points second
age_pts_df = player_df.sort_values(by=['age', 'pts'], ascending=[False, True])
display(age_pts_df)

# sort the subsetted dataframe for high scorers by average rebounds and assists per game
high_scorers_reb = high_scorers.sort_values(by=["reb", "ast"], ascending=False)
display(high_scorers_reb)

player_name team_abbreviation   age  player_height  player_weight  \
4844      Kevin Willis               DAL  44.0         213.36     111.130040   
5436   Dikembe Mutombo               HOU  43.0         218.44     117.933920   
230      Robert Parish               CHI  43.0         215.90     110.676448   
12390    Udonis Haslem               MIA  43.0         200.66     106.594120   
11032     Vince Carter               ATL  43.0         198.12      99.790240   
...                ...               ...   ...            ...            ...   
3448      LeBron James               CLE  19.0         203.20     108.862080   
11029  Zion Williamson               NOP  19.0         198.12     128.820128   
4333      Andrew Bynum               LAL  18.0         213.36     124.737800   
72     Jermaine O'Neal               POR  18.0         210.82     102.511792   
334        Kobe Bryant               LAL  18.0         200.66      90.718400   

      country draft_year draft_round  gp   pts  reb  ast  
4844      USA       1984           1   5   2.4  1.6  0.2  
5436    Congo       1991           1   9   1.8  3.7  0.0  
230       USA       1976           1  43   3.7  2.1  0.5  
12390     USA  Undrafted   Undrafted   7   3.9  1.6  0.0  
11032     USA       1998           1  60   5.0  2.1  0.8  
...       ...        ...         ...  ..   ...  ...  ...  
3448      USA       2003           1  79  20.9  5.5  5.9  
11029     USA       2019           1  24  22.5  6.3  2.1  
4333      USA       2005           1  46   1.6  1.7  0.2  
72        USA       1996           1  45   4.1  2.8  0.2  
334       USA       1996           1  71   7.6  1.9  1.3  

[12844 rows x 12 columns]

player_name team_abbreviation   age  player_height  player_weight  \
6292        Kevin Love               MIN  22.0         208.28     117.933920   
6904     Dwight Howard               ORL  26.0         210.82     120.201880   
4918     Dwight Howard               ORL  22.0         210.82     120.201880   
6392     Dwight Howard               ORL  25.0         210.82     120.201880   
3434     Kevin Garnett               MIN  28.0         210.82     108.862080   
...                ...               ...   ...            ...            ...   
9537     Isaiah Thomas               BOS  28.0         175.26      83.914520   
12807     Jordan Poole               GSW  24.0         193.04      87.996848   
4987    Jamal Crawford               NYK  28.0         195.58      90.718400   
12586  Anfernee Simons               POR  24.0         190.50      82.100152   
9667      Lou Williams               LAC  31.0         185.42      79.378600   

      country draft_year draft_round  gp   pts   reb  ast  
6292      USA       2008           1  73  20.2  15.2  2.5  
6904      USA       2004           1  54  20.6  14.5  1.9  
4918      USA       2004           1  82  20.7  14.2  1.3  
6392      USA       2004           1  78  22.9  14.1  1.4  
3434      USA       1995           1  82  24.2  13.9  5.0  
...       ...        ...         ...  ..   ...   ...  ...  
9537      USA       2011           2  76  28.9   2.7  5.9  
12807     USA       2019           1  82  20.4   2.7  4.5  
4987      USA       2000           1  80  20.6   2.6  5.0  
12586     USA       2018           1  62  21.1   2.6  4.1  
9667      USA       2005           2  79  22.6   2.5  5.3  

[701 rows x 12 columns]

The first sort helps to identify the players that are old yet high scorers. It also helps identify the players who are younger and still had a lot of points.

The second sort helps visualize who the top scorers mixed with the most rebounds and assists per game are. Essentially, telling us who valuable players on teams were (MVP).

In [22]:
full_df = full_df[["season", "player_id", "player", "pos", "age", "experience", "lg", "tm", "g", "mp", "per"]]
display(full_df)

season  player_id          player  pos   age  experience   lg   tm   g  \
0        2024       5025      A.J. Green   SG  24.0           2  NBA  MIL  11   
1        2024       5027      AJ Griffin   SF  20.0           2  NBA  ATL  11   
2        2024       4219    Aaron Gordon   PF  28.0          10  NBA  DEN  16   
3        2024       4582   Aaron Holiday   PG  27.0           6  NBA  HOU  14   
4        2024       4805   Aaron Nesmith   SF  24.0           4  NBA  IND  15   
...       ...        ...             ...  ...   ...         ...  ...  ...  ..   
31628    1947        157     Walt Miller    F  31.0           1  BAA  PIT  12   
31629    1947        158   Warren Fenley    F  24.0           1  BAA  BOS  33   
31630    1947        159   Wilbert Kautz  G-F  31.0           1  BAA  CHS  50   
31631    1947        160  Woody Grimshaw    G  27.0           1  BAA  PRO  21   
31632    1947        161     Wyndol Gray  G-F  24.0           1  BAA  BOS  55   

          mp   per  
0       81.0   9.0  
1      101.0   3.4  
2      542.0  14.2  
3      258.0  10.5  
4      351.0  14.1  
...      ...   ...  
31628    NaN   NaN  
31629    NaN   NaN  
31630    NaN   NaN  
31631    NaN   NaN  
31632    NaN   NaN  

[31633 rows x 11 columns]

In [23]:
# remove missing values for first dataset
full_df["mp"].fillna(0, inplace=True)
full_df["per"].fillna(8, inplace=True)
display(full_df)

season  player_id          player  pos   age  experience   lg   tm   g  \
0        2024       5025      A.J. Green   SG  24.0           2  NBA  MIL  11   
1        2024       5027      AJ Griffin   SF  20.0           2  NBA  ATL  11   
2        2024       4219    Aaron Gordon   PF  28.0          10  NBA  DEN  16   
3        2024       4582   Aaron Holiday   PG  27.0           6  NBA  HOU  14   
4        2024       4805   Aaron Nesmith   SF  24.0           4  NBA  IND  15   
...       ...        ...             ...  ...   ...         ...  ...  ...  ..   
31628    1947        157     Walt Miller    F  31.0           1  BAA  PIT  12   
31629    1947        158   Warren Fenley    F  24.0           1  BAA  BOS  33   
31630    1947        159   Wilbert Kautz  G-F  31.0           1  BAA  CHS  50   
31631    1947        160  Woody Grimshaw    G  27.0           1  BAA  PRO  21   
31632    1947        161     Wyndol Gray  G-F  24.0           1  BAA  BOS  55   

          mp   per  
0       81.0   9.0  
1      101.0   3.4  
2      542.0  14.2  
3      258.0  10.5  
4      351.0  14.1  
...      ...   ...  
31628    0.0   8.0  
31629    0.0   8.0  
31630    0.0   8.0  
31631    0.0   8.0  
31632    0.0   8.0  

[31633 rows x 11 columns]

In [28]:
# subsetting the dataset


AttributeError: 'DataFrameGroupBy' object has no attribute 'groupby'